doc/contributing/coding-conventions.chapter.md

```diff
@@ -181,6 +181,21 @@
   rev = "${version}";
   ```
 
+- Filling lists condionally _should_ be done with `lib.optional(s)` instead of using `if cond then [ ... ] else null`.
```

Comment:
```suggestion
- Filling lists condionally _should_ be done with `lib.optional(s)` instead of using `if cond then [ ... ] else []`.
```
This is the implementation of optional(s) and will works with list operators (like `++`).

Reply:
The null is intentionally to empathies the build avoiding tricks.

Reply:
I don't get why focusing on the null case.
On master I see 6 occurrences of null versus 177 occurrences of [].

```
pcregrep -rM "]\s+else\s+null" . | wc -l
pcregrep -rM "]\s+else\s+\[\s*\]" . | wc -l
```

Reply:
Because otherwise the surrounding sentence loses context but we can add both.

```suggestion
- Filling lists condionally _should_ be done with `lib.optional(s)` instead of using `if cond then [ ... ] else null` or `if cond then [ ... ] else [ ]`.
```